### ISSUE TITLE
Installation fails on Ubuntu 18.04 "The repository 'https://apt.dockerproject.org/repo ubuntu-bionic InRelease' is not signed."

### ISSUE THREAD

**Issue by garethhowell** [saltstack-formulas/docker-formula#202]
```
docker1:
----------
          ID: docker package dependencies
    Function: pkg.installed
      Result: True
     Comment: All specified packages are already installed
     Started: 16:22:47.810360
    Duration: 57.869 ms
     Changes:   
----------
          ID: purge old packages
    Function: pkgrepo.absent
        Name: deb https://get.docker.com/ubuntu docker main
      Result: True
     Comment: Package repo deb https://get.docker.com/ubuntu docker main is absent
     Started: 16:22:47.869817
    Duration: 44.893 ms
     Changes:   
----------
          ID: purge old packages
    Function: pkg.purged
      Result: True
     Comment: All specified packages are already absent
     Started: 16:22:47.915298
    Duration: 21.814 ms
     Changes:   
----------
          ID: docker package repository
    Function: pkgrepo.managed
        Name: deb https://apt.dockerproject.org/repo ubuntu-bionic main
      Result: False
     Comment: Failed to configure repo 'deb https://apt.dockerproject.org/repo ubuntu-bionic main': E: Failed to fetch https://apt.dockerproject.org/repo/dists/ubuntu-bionic/InRelease  403  Forbidden [IP: 54.192.10.189 443]
              E: The repository 'https://apt.dockerproject.org/repo ubuntu-bionic InRelease' is not signed.
     Started: 16:22:47.937995
    Duration: 268613.387 ms
     Changes:   
----------
          ID: docker-config
    Function: file.managed
        Name: /etc/default/docker
      Result: True
     Comment: File /etc/default/docker is in the correct state
     Started: 16:27:16.558617
    Duration: 295.526 ms
     Changes:   
----------
          ID: docker package
    Function: pkg.latest
        Name: docker-engine
      Result: False
     Comment: One or more requisite failed: docker.docker package repository
     Started: 16:27:16.855007
    Duration: 0.021 ms
     Changes:   
----------
          ID: docker-service
    Function: service.running
        Name: docker
      Result: False
     Comment: One or more requisite failed: docker.docker package
     Started: 16:27:16.856687
    Duration: 0.024 ms
     Changes:   

Summary for docker1
------------
Succeeded: 4
Failed:    3
------------
Total states run:     7
Total run time: 269.034 s
```

**Comment by garethhowell** [comment]
The problem is that apt.dockerproject.org does not contain a distribution for ubuntu-bionic.

**Comment by myii** [comment]
This a potential solution: https://download.docker.com/linux/ubuntu/dists/bionic/pool/stable/amd64/.

Taken from https://linuxconfig.org/how-to-install-docker-on-ubuntu-18-04-bionic-beaver:

> ```
> STABLE:
> deb [arch=amd64] https://download.docker.com/linux/ubuntu bionic stable
> EDGE:
> deb [arch=amd64] https://download.docker.com/linux/ubuntu bionic edge
> NIGHTLY:
> deb [arch=amd64] https://download.docker.com/linux/ubuntu bionic nightly
> ```

**Comment by garethhowell** [comment]
Thanks.
I guess the real question is “why is the formula downloading from
apt.dockerproject.org. rather than download,docker.com in the first place.”


Gareth

**Comment by martinadolfi** [comment]
ASMOF, in `osfamilymap.yaml` has the `download.docker.com` repos defined that sould work:
```
    url_base: https://download.docker.com/linux/{{ grains['os'] |lower }}
```
I think that should be available as `docker.repo.url_base` in `repo.sls`  as also some other parts of that object is used in  `- key_url: {{ docker.repo.key_url }}` on line 23

I think the problem is that on `repo.sls` the `set url` defined on line 11 sets again the url based on other stuff. 

Perhaps a interesting solution would be to build the url based on `url_base`.

I'll try it now and let you know if it works (for bionic, of course it'll need more testing)

**Comment by martinadolfi** [comment]
It worked properly on bionic. This is the change I made, just one line>
```
root@srv-salt-master-1:/srv/formulas/docker-formula# git diff
diff --git a/docker/repo.sls b/docker/repo.sls
index 61a84f2..88160da 100644
--- a/docker/repo.sls
+++ b/docker/repo.sls
@@ -8,7 +8,7 @@
 {% set humanname_old = 'Old ' if docker.use_old_repo else '' %}

 {%- if grains['os_family']|lower in ('debian',) %}
-{% set url = 'https://apt.dockerproject.org/repo ' ~ grains["os"]|lower ~ '-' ~ grains["oscodename"] ~ ' main' if docker.use_old_repo else docker.repo.url_base ~ ' ' ~ docker.repo.version ~ ' stable' %}
+{% set url = docker.repo.url_base ~ ' ' ~ grains["oscodename"]|lower ~ ' stable' %}

 docker-package-repository:
   pkgrepo.{{ repo_state }}:
root@srv-salt-master-1:/srv/formulas/docker-formula#
```
Does it make sense? 
I'll try later with other versions

**Comment by garethhowell** [comment]
Thanks, Martin
I’ll test that out on Monday and report back.

Gareth

**Comment by noelmcloughlin** [comment]
Fixed in merged #256
